src/ext/ddtrace.c

```diff
@@ -48,6 +50,130 @@
 #define PHP7_UNUSED(...) UNUSED(__VA_ARGS__)
 #endif
 
+/* {{{ arginfo */
```

Comment:
Thanks for the CR! I'm thinking of reverting moving any of the interfaces to the C level and changing the `dd_trace_serialize_trace()` API to accept a plain-old PHP array instead of an instance of the tracer. What do you think? :)